src/util/config.rs

```
@@ -0,0 +1,164 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use rocksdb::DBCompressionType;
+
+pub fn get_compression_by_string(tp: &str) -> DBCompressionType {
+    match &*tp.to_owned().to_lowercase() {
+        "no" => DBCompressionType::DBNo,
+        "snappy" => DBCompressionType::DBSnappy,
+        "zlib" => DBCompressionType::DBZlib,
+        "bzip2" => DBCompressionType::DBBz2,
+        "lz4" => DBCompressionType::DBLz4,
+        "lz4hc" => DBCompressionType::DBLz4hc,
+        _ => panic!("unsupported compression type {}", tp),
+    }
+}
+
+fn split_property(property: &str) -> Result<(f64, &str), (())> {
+    let mut indx = 0;
+    for s in property.chars() {
+        match s {
+            '0'...'9' | '.' => {
+                indx += 1;
+            }
+            _ => {
+                break;
+            }
+        }
+    }
+
+    let (num, unit) = property.split_at(indx);
+    match num.parse::<f64>() {
+        Ok(f) => Ok((f, unit)),
+        Err(_) => Err(()),
+    }
+}
+
+const UNIT: i64 = 1;
+const DATA_MAGNITUDE: i64 = 1024;
+const KB: i64 = UNIT * DATA_MAGNITUDE;
+const MB: i64 = KB * DATA_MAGNITUDE;
+const GB: i64 = MB * DATA_MAGNITUDE;
+// When we cast TB and PB to `f64`, compiler will report a overflow warning.
+// TB = GB * DATA_MAGNITUDE
+// PB = PB * DATA_MAGNITUDE
+
+const TIME_MAGNITUDE_1: i64 = 1000;
+const TIME_MAGNITUDE_2: i64 = 60;
+const MS: i64 = UNIT;
+const SECOND: i64 = MS * TIME_MAGNITUDE_1;
+const MINTUE: i64 = SECOND * TIME_MAGNITUDE_2;
+const HOUR: i64 = MINTUE * TIME_MAGNITUDE_2;
+
+pub fn get_integer_by_string(size: &str) -> Result<i64, ()> {
```

please return a parse error here.